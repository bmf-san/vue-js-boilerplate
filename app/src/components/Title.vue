<template>
  <div class="title">
    <h1>{{ msg }}</h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class Title extends Vue {
    @Prop() private msg!: string;
};
</script>
