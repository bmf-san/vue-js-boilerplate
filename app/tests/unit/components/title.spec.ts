import { shallowMount } from "@vue/test-utils";
import Title from "@/components/Title.vue";

describe("Title.vue", () => {
  it("init", () => {
    const wrapper = shallowMount(Title);
    expect(wrapper.props().msg).toBeUndefined();
    expect(wrapper.isVueInstance).toBeTruthy();
  });
  
  it("renders props.msg when passed", () => {
    const msg = "new message";
    const wrapper = shallowMount(Title, {
      propsData: { msg }
    });
    expect(wrapper.text()).toMatch(msg);
  });

  it("matched snapshot", () => {
    const wrapper = shallowMount(Title);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
