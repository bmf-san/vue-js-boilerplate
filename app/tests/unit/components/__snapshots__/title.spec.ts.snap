// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title.vue matched snapshot 1`] = `
<div class="title">
  <h1></h1>
</div>
`;
